src/main/java/com/uid2/admin/legacy/LegacyClientKeyStoreWriter.java

```diff
@@ -24,6 +26,18 @@ public LegacyClientKeyStoreWriter(RotatingLegacyClientKeyProvider provider, File
         String dataType = "client_keys";
         writer = new ScopedStoreWriter(provider, fileManager, versionGenerator, clock, scope, dataFile, dataType);
     }
+    public LegacyClientKeyStoreWriter(RotatingLegacyClientKeyProvider provider,
+                                      FileManager fileManager,
+                                      ObjectWriter jsonWriter,
+                                      VersionGenerator versionGenerator,
+                                      Clock clock,
+                                      StoreScope scope,
+                                      RotatingS3KeyProvider s3KeyProvider) {
+        this.jsonWriter = jsonWriter;
+        FileName dataFile = new FileName("clients", ".json");
+        String dataType = "client_keys";
+        this.writer = new EncryptedScopedStoreWriter(provider, fileManager, versionGenerator, clock, scope, dataFile, dataType, s3KeyProvider, scope.getId());
```

Comment:
Are you 100% sure this will never be a globalscope? That returns a null for getId().

Reply:
Global scope would only be passed into this legacyClientKeyStoreWriter constructor during main and globalWriter in dataTypeFactories. I have checked and don't think globalScope would be passed in with this constructor having S3 key provider by any chance. This constructor with s3 key provider is only called in dataTypeFactories getEncryptedWriter function, which passed in encryptedScope only. I will change this constructor from taking ScopedStore to EncryptedScopeStorage, in case there are any mismatch that I missed.

Reply:
Now all the constructors that will initiate EncryptedScopedStoreWriter takes in EncryptedScope instead of ScopedStore. As long as the program runs without errors, that should verify no globalScope indeed come at this constuctor.